sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -45,7 +45,10 @@ class SparkSqlParser(conf: SQLConf) extends AbstractSqlParser(conf) {
   private val substitutor = new VariableSubstitution(conf)
 
   protected override def parse[T](command: String)(toResult: SqlBaseParser => T): T = {
-    super.parse(substitutor.substitute(command))(toResult)
+    val trimmed_command = if (command.trim.endsWith(";")) {
```

Comment:
Can we fix this in the `SqlBase.g4` side? Then, add tests in `PlanParserSuite`?

Reply:
Yeah that would actually be much cleaner. What do you think about adding the quantifier as `(';')*?` to allow for any number of semicolons, i.e.
```scala
sql("select name from people;;;;;;")
```
It goes beyond handling the accidental additional semicolon but I don't see why any number of semicolons at the end would make a query ambiguous.

Reply:
@maropu I updated the PR as described including some tests, let me know what you think. To expand on the example in my earlier reply:

```scala
// Below query is understood to be a single statement
sql("select 1, 2, 3;; ;; ; ")
res1: org.apache.spark.sql.DataFrame = [1: int, 2: int ... 1 more field]

// Below multi-statement query appropriately fails
sql("select 1, 2, 3;; ;; select 4, 5, 6; ")
org.apache.spark.sql.catalyst.parser.ParseException:
extraneous input 'select' expecting {<EOF>, ';'}(line 1, pos 20)

// == SQL ==
// select 1, 2, 3;; ;; select 4, 5, 6
// --------------------^^^
```

Reply:
How about this new behaviour? That looks fine to me though. @cloud-fan @dongjoon-hyun @HyukjinKwon